src/Chapter1.hs

```diff
@@ -635,8 +646,8 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n = if div (abs n) 10 == 0 then n else firstDigit (div (abs n) 10)
```

Comment:
💪🏼 Nice one!

A few things in here to pay attention to. You can see that `div (abs n) 10` is used twice in this function. In Haskell, the function with the same input will always produce the same result. That means, that these two expressions are always the same. So you can move in into the separate variable through the `let-in` or `where` constructions 🙂

Reply:
Also, I can see that there is a small nuance for numbers from -10 to 0 (negative one-digits). We apply the `div` function to the absolute value (which is _absolute_ correct), however, when we return the value, we return `n` itself. That would include minus as well. So this could be improved a bit 🙂